src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -1113,17 +1113,17 @@ static std::vector<TestCollection> testSuite = {
  * print basic information about a test
  */
 static void
-pretty_print_test_desc(const boost::property_tree::ptree& test, int test_idx,
-                       size_t testSuiteSize, std::ostream & _ostream, const std::string& bdf)
+pretty_print_test_desc(const boost::property_tree::ptree& test, int& test_idx,
```

Comment:
How the variable `test_index` is being used here is unfortunately a kludge, please don't do this.  

In addition how this function is being used doesn't look right.  When you get a chance, please ping me so that we can go over this code.

Reply:
The refactoring requires a lot more work than anticipated. We will revisit this after finishing the U30 work